gears/src/types/base/coins/unsigned.rs

```diff
@@ -26,6 +28,68 @@ impl UnsignedCoins {
 
         true
     }
+
+    // TODO: Move this to generic declaration
+    /// Substracts matching coins. If the other coins have bigger values or the coins that don't
+    /// exists in original set, method returns error. If all coins are identical method returns
+    /// error.
+    pub fn checked_sub(&self, other: &UnsignedCoins) -> Result<Self, CoinsError> {
```

Comment:
Is there a test for this method?

Reply:
This method is a copy from  decimal implementation done by @dmvict , and it has tests. I made a quick copy for now of his code.